champsim_config.json

```diff
@@ -159,19 +159,21 @@
     },
 
     "physical_memory": {
-        "frequency": 3200,
+        "data_rate": 3200,
         "channels": 1,
         "ranks": 1,
         "banks": 8,
         "rows": 65536,
-        "columns": 128,
+        "columns": 1024,
```

Comment:
This change is pretty substantial, and I think it's worth at least considering changing the key so that the different behavior doesn't rug pull someone. Could this be `"column_width"`?

Reply:
I think column_width implies that its the addressing bits width we are defining, which would be misleading.
I think we could change rows and columns to bank_rows and bank_columns which would be accurate.

I think if the config file specified a unit for all its numerical values, like a number followed by some string indicating the unit, that would resolve issues like this in the future as well. So users could define 1024b for bits or 128B for bytes. We could then also take care of the units not being clear from the config file. I actually really like this idea.

Reply:
Something like [this](https://github.com/ChampSim/ChampSim/blob/33dca7a859a273b04da10784b25faafc2acdfca8/config/parse.py#L93)?

Reply:
I realize it isn't exactly what you're looking for, but the bones are there. I think `bank_rows` and `bank_columns` communicate better that these are rows per bank and columns per bank, if you don't already know that.

Reply:
Yeah. I would like it to work for all of ChampSim's types if possible (and have a way to catch incompatible type conversions if possible). So instead of just bytes, also times, bits, and cycles.
I don't think that would be too difficult to implement, but it will take some tedious changes to the config scripts.